src/atomate2/forcefields/schemas.py

```diff
@@ -0,0 +1,167 @@
+"""Job to prerelax a structure using an MD Potential."""
+
+import logging
+from typing import List
+
+from pydantic import BaseModel, Field
+
+# from jobflow import Maker, job
+from pymatgen.core.structure import Structure
+
+
+class InputDoc(BaseModel):
+    """The inputs used to run this job."""
+
+    structure: Structure = Field(None, description="The inputted structure.")
+    relax_cell: bool = Field(
+        None,
+        description="Whether cell lattice was allowed to change during relaxation.",
+    )
+    relax_kwargs: dict = Field(
+        None,
+        description="Keyword arguments that passed to :obj:`StructOptimizer.relax`.",
+    )
+    optimizer_kwargs: dict = Field(
+        None,
+        description="Keyword arguments that passed to :obj:`StructOptimizer`.",
+    )
+
+
+class OutputDoc(BaseModel):
+    """The outputs of this job."""
+
+    structure: Structure = Field(None, description="The final, relaxed structure.")
+
+    energy: float = Field(None, description="Total energy in units of eV.")
+
+    energy_per_atom: float = Field(
+        None,
+        description="Energy per atom of the final structure in units of eV/atom.",
+    )
+
+    forces: List[List[float]] = Field(
+        None,
+        description="The force on each atom in units of eV/A for the final structure.",
+    )
+
+    # NOTE: units for stresses were converted to kbar
+    # (* -10 from standard output) to comply with MP convention
+    stress: List[float] = Field(
+        None, description="The stress on the cell in units of kbar (in Voigt notation)."
+    )
+
+    trajectory: dict = Field(
+        None, description="Step-by-step trajectory of the structural relaxation."
+    )
+
+    steps: int = Field(
+        None, description="total number of steps needed to relax the structure."
+    )
+
+
+class FFStructureRelaxDocument(BaseModel):
```

Comment:
Can you make this subclass StructureMetaData like the VASP task document does. This will add a load of useful metadata (like formula, volume, composition, etc) to the document automatically.

Reply:
Currently, when I try to make this a subclass of StructureMetadata, I get the error 

> Traceback (most recent call last):
  File "/Users/kuner/opt/anaconda3/envs/atomate2/lib/python3.9/site-packages/jobflow/core/job.py", line 587, in run
    output = jsanitize(response.output, strict=True, enum_values=True)
  File "/Users/kuner/opt/anaconda3/envs/atomate2/lib/python3.9/site-packages/monty/json.py", line 559, in jsanitize
    return jsanitize(
  File "/Users/kuner/opt/anaconda3/envs/atomate2/lib/python3.9/site-packages/monty/json.py", line 526, in jsanitize
    return {
  File "/Users/kuner/opt/anaconda3/envs/atomate2/lib/python3.9/site-packages/monty/json.py", line 527, in <dictcomp>
    str(k): jsanitize(
  File "/Users/kuner/opt/anaconda3/envs/atomate2/lib/python3.9/site-packages/monty/json.py", line 526, in jsanitize
    return {
  File "/Users/kuner/opt/anaconda3/envs/atomate2/lib/python3.9/site-packages/monty/json.py", line 527, in <dictcomp>
    str(k): jsanitize(
  File "/Users/kuner/opt/anaconda3/envs/atomate2/lib/python3.9/site-packages/monty/json.py", line 568, in jsanitize
    obj.as_dict(),
AttributeError: 'datetime.datetime' object has no attribute 'as_dict'

> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/Users/kuner/opt/anaconda3/envs/atomate2/lib/python3.9/site-packages/jobflow/managers/local.py", line 98, in _run_job
>     response = job.run(store=store)
>   File "/Users/kuner/opt/anaconda3/envs/atomate2/lib/python3.9/site-packages/jobflow/core/job.py", line 589, in run
>     raise RuntimeError(
> RuntimeError: Job output contained an object that is not MSONable and therefore could not be serialized.

Any ideas?

Reply:
The serialization handler needs another check:
```py
if isinstance(obj, datetime. datetime):
   return str(obj)
```

Maybe check if atomate2 uses `str` or some other serialization function.